crates/forge/src/compatibility_check.rs

```diff
@@ -52,15 +53,29 @@ impl<'a> RequirementsChecker<'a> {
         for requirement in &self.requirements {
             let raw_version = get_raw_version(&requirement.name, &requirement.command)?;
             let version = (requirement.version_parser)(&raw_version)?;
-            let is_valid = version >= requirement.minimal_version;
-            let command_output = if is_valid {
-                format!("✅ {} {}", requirement.name, version)
+            let is_valid = if let Some(minimal_recommended_version) =
+                &requirement.minimal_recommended_version
+            {
+                minimal_recommended_version <= &version
             } else {
-                all_valid = false;
-                format!(
-                    "❌ {} Version {} doesn't satisfy minimum {}\n{}",
-                    requirement.name, version, requirement.minimal_version, requirement.helper_text
-                )
+                requirement.minimal_version <= version
+            };
+            let command_output = match (is_valid, &requirement.minimal_recommended_version) {
+                (true, _) => format!("✅ {} {}", requirement.name, version),
+                (false, Some(minimal_recommended_version)) => format!(
+                    "⚠️  {} Version {} doesn't satisfy minimum recommended {}\n{}",
+                    requirement.name, version, minimal_recommended_version, requirement.helper_text
+                ),
+                (false, None) => {
+                    all_valid = false;
+                    format!(
+                        "❌ {} Version {} doesn't satisfy minimum {}\n{}",
+                        requirement.name,
+                        version,
+                        requirement.minimal_version,
+                        requirement.helper_text
+                    )
+                }
```

Comment:
I think it doesn't correctly handle the case where user have version below recommended installed.

For me, the expected behavior would be to show "Version ... doesn't satisfy minimum...` but I think your logic will still show `Version {} doesn't satisfy minimum recommended`.

Reply:
Unit tests for these cases should be added to the file

Reply:
> I think it doesn't correctly handle the case where user have version below recommended installed.
It does, e.g. if `minimal <= our version < minimal recommended`, we will get a warning.

> Unit tests for these cases should be added to the file
Added